### PR TITLE
update service definitions

### LINKED ISSUE #2: Node info should store datetime values in ISO format

### LINKED ISSUE #3: Proposed new format
- Add "date_added" field (iso format datetime) in node_registry.json (indicates when the node was added to the network)
- Remove "deployed since" from node_info.json
- Remove "location" (that info can be added to affiliation if needed)
- Replace "administrators" and "support_contact" with "contact" 
- Add "date_updated" field (iso format datetime) in node_info.json (indicates when the node info was last changed)
- Add "available" field (boolean) in node_info.json (indicates whether the node responds when updating info)
- Replace "capabilities" field with "services" (list of services available at a given node and the relative path)

node_registry.json
```json
{
  "name": {"url": "example.com", "date_added": "....."}
}
```

node_info.json
```json
{
    "name":{
        "affiliation": ".....",
        "services": {
            "thredds": "/thredds",
            "catalog": "/stac",
            "jupyter": "/jupyter/hub"
        },
        "date_updated": "........",
        "contact": "somesupport@example.com"
    },
}
```

### PR DESCRIPTION
**Pull request by mishaschwartz**
Update service definitions to include the following keys only:

- name
- keywords (replaces `types`)
- description
- links (schema defined by `https://json-schema.org/draft/2020-12/links` with the additional restriction that there must be at least one `"rel"` key with the value `"self"` and one with the valued `"service-doc"`)

`"service"` contains an "href" that refers to the context URI of the service (ie. what the `"url"` key used to refer to)
`"service-doc"` contains an "href" that refers to the documentation of the service (ie. what the `"documentation"` key used to refer to)

See https://www.iana.org/assignments/link-relations/link-relations.xhtml for link relation definitions.

Resolves #2 
Resolves #3

### CODE REVIEW

**Review by fmigneault — COMMENTED** [review]
Relations `self` should be replaced by `service`. 

The logic behind `self` when used within a nested document is to have the link that allows access to this nested definition by itself. So, for example, accessing `/services` returns the list of all service items, each with a `self` that points at `/services/{service-id}`, returning only that service definition.
The `service` relation better represents what is provided as URL, aka where the *actal* service being described here is located.

> https://www.iana.org/assignments/link-relations/link-relations.xhtml
> `service` | Indicates a URI that can be used to retrieve a service document.

Probably also a good idea to ensure `links` schema has `contains` this `service` relationship as well.
The rest look fine.

**Comment by fmigneault** [comment]
Another thing that could be nice. 
Having `links` directly under each node with `collection` rel that points at eg: `http://daccs-uoft.example.com/services`. 
We could also consider multiple other utility rel, such as `about`, `describes`, `status` etc. to refer to the node's documentation references as well if available (this can be done in subsequent PRs).

File: node_registry.schema.json [hunk 1]
```diff
@@ -1,100 +1,136 @@
...
+          "type": "string"
+        },
+        "description": {
+          "type": "string"
+        },
+        "icon_url": {
+          "type": "string",
+          "format": "uri",
+          "pattern": "^https?://"
+        },
```
Comment (fmigneault) [inline node_registry.schema.json:26]: See `icon` relationship name.

Comment (mishaschwartz) [reply node_registry.schema.json:26]: This isn't part of the `links` list since it refers to an optional icon for the whole node, not a single service.

Also, this value is set manually when registering the node so it's easier for the user to add/update in this format.

File: tests/fixtures/links-schema-cache.json [hunk 2]
```diff
@@ -0,0 +1,85 @@
+{
```
Comment (fmigneault) [inline tests/fixtures/links-schema-cache.json:1]: Is this file still necessary since the original source is referenced directly in the schema: https://github.com/DACCS-Climate/DACCS-node-registry/pull/9/files#diff-2c4b9d4938eba1be2882c99ad66879ed9c28d8815f429b5aece574ddb6efad79R76-R77 ?

Comment (fmigneault) [reply tests/fixtures/links-schema-cache.json:1]: I saw the offline test after, so I guess its fine. 
Does it actually work? That schema uses `"$dynamicRef": "https://json-schema.org/draft/2020-12/hyper-schema#meta"`, so it should still attempt an online resolution anyway?

Comment (mishaschwartz) [reply tests/fixtures/links-schema-cache.json:1]: Yes, it's a bad idea to make your tests dependent on an internet connection. See here:

https://github.com/DACCS-Climate/DACCS-node-registry/pull/9/files#diff-aa57560853f73b4713b47a1ae27f28c76a227216db2eb068e32649322876ae94R45-R47

File: node_registry.schema.json (hunk 1 above)
Comment (mishaschwartz) [reply node_registry.schema.json:26]: To consider in a later PR

**Comment by mishaschwartz** [comment]
> Another thing that could be nice....

Thanks, we'll consider that for a future PR

File: tests/fixtures/links-schema-cache.json (hunk 2 above)
Comment (mishaschwartz) [reply tests/fixtures/links-schema-cache.json:1]: Yes it actually works. Dynamic-refs are only resolved if the property is actually present. Since we're currently only testing links with rel, type, and href; the dynamic-ref is never triggered. 

If we ever change the tests to include some of those additional properties then, yes, we'd have to mock those ones as well.

**Comment by dchandan** [comment]
All good :)
